projects/react/src/lib/useMaskito.ts

```diff
@@ -32,26 +33,32 @@ export const useMaskito = ({
     options?: MaskitoOptions;
     elementPredicate?: MaskitoElementPredicate;
 } = {}): RefCallback<HTMLElement> => {
-    const [element, setElement] = useState<HTMLElement | null>(null);
+    const [predicateResult, setPredicateResult] = useState<
+        HTMLInputElement | HTMLTextAreaElement | null
+    >(null);
 
     const onRefChange: RefCallback<HTMLElement> = useCallback(
-        (node: HTMLElement | null) => {
-            setElement(node);
+        async (node: HTMLElement | null) => {
+            if (node) {
+                const predicate = await elementPredicate(node);
+
+                setPredicateResult(predicate);
```

Comment:
@nextZed Could you review changes inside this file (`@maskito/react`), please ?

Reply:
I think we can't make ref callback async, because ref callbacks run synchronously in commit phase. I suggest to separate predicates typings 
```ts
export type SyncMaskitoElementPredicate<T extends HTMLElement = HTMLElement> = (
    element: T,
) =>
    | HTMLInputElement
    | HTMLTextAreaElement

export type AsyncMaskitoElementPredicate<T extends HTMLElement = HTMLElement> = (
    element: T,
) => Promise<HTMLInputElement | HTMLTextAreaElement>;

export type MaskitoElementPredicate<T extends HTMLElement = HTMLElement> = SyncMaskitoElementPredicate<T> | AsyncMaskitoElementPredicate<T>
``` 

and use only `SyncMaskitoElementPredicate` in `@maskito/react`